packages/core/ui-settings/core-ui-settings-server-internal/src/ui_settings_service.test.ts

```diff
@@ -275,5 +278,63 @@ describe('uiSettings', () => {
         expect(MockUiSettingsGlobalClientConstructor.mock.calls[0][0].defaults).toEqual({});
       });
     });
+
+    describe('#asScopedToUserClient', () => {
+      it('passes saved object type "config-user" to UiSettingsUserClient', async () => {
+        await service.setup(setupDeps);
+        const start = await service.start();
+        start.userAsScopedToClient(savedObjectsClient);
+
+        expect(MockUiSettingsUserClientConstructor).toBeCalledTimes(1);
+        expect(MockUiSettingsUserClientConstructor.mock.calls[0][0].type).toBe('config-user');
+      });
+
+      it('passes overrides to UiSettingsUserClient', async () => {
+        await service.setup(setupDeps);
+        const start = await service.start();
+        start.userAsScopedToClient(savedObjectsClient);
+
+        expect(MockUiSettingsUserClientConstructor).toBeCalledTimes(1);
+        expect(MockUiSettingsUserClientConstructor.mock.calls[0][0].overrides).toEqual({});
+      });
+
+      it('passes a copy of set defaults to UiSettingsUserClient', async () => {
+        const setup = await service.setup(setupDeps);
+        setup.register(defaults);
+        const start = await service.start();
+        start.userAsScopedToClient(savedObjectsClient);
+
+        expect(MockUiSettingsUserClientConstructor).toBeCalledTimes(1);
+        expect(MockUiSettingsUserClientConstructor.mock.calls[0][0].defaults).toEqual({});
+      });
+    });
+
+    describe('#setUserProfileSettingsClientFactoryProvider', () => {
+      it('sets the UserProfileSettingsClientFactoryProvider', async () => {
+        await service.setup(setupDeps);
+        const start = await service.start();
+
+        const userProfileSettingsClientMock = {
+          // @ts-ignore
```

Comment:
we shouldn't be using `ts-ignore`. what is the error you are trying to avoid here?

Reply:
Great catch! My mock needed to take in a `request` but it doesn't use it. I was trying to avoid an 'Unused Argument' check

Reply:
Fixed in latest commit!